src/main.ts

```diff
@@ -105,16 +120,25 @@ const router = createRouter({
   ],
 });
 
+router.beforeEach((to, from, next) => {
+  if (
+    from.params.lang === to.params.lang ||
+    !languageCodesLowerCase.includes(to.params.lang as string)
+  ) {
+    next();
+    return;
+  }
+  if (languageCodesLowerCase.includes(to.params.lang as string)) {
+    i18n.global.locale.value =
+      languageCodes[
+      languageCodesLowerCase.indexOf(to.params.lang as string)
+      ];
+  }
+  next();
```

Comment:
This is, in my opinion, much simpler. What do you think?
```suggestion
  const toLang = to.params.lang as string;
  if (from.params.lang !== toLang && languageCodesLowerCase.includes(toLang)) {
    i18n.global.locale.value = languageCodes[languageCodesLowerCase.indexOf(toLang)];
  }
  next();
```